File: core/src/com/mirage/view/scene/objects/humanoid/StaticTexture.java
package com.mirage.view.scene.objects.humanoid;

import com.badlogic.gdx.graphics.Texture;

public class StaticTexture extends AnimatedTexture {
    private Texture texture;

    public StaticTexture(Texture texture) {
        this.texture = texture;
    }

    @Override
    public Texture getTexture(long timePassed) {
        return texture;
    }
}

File: core/src/com/mirage/view/scene/objects/humanoid/StaticTexture.kt
package com.mirage.view.scene.objects.humanoid

import com.badlogic.gdx.graphics.Texture
import com.mirage.view.scene.objects.Image

/**
 * Статичная текстура, не изменяющаяся со временем
 */
class StaticTexture(private var t: Texture) : Image() {

    override fun getTexture(): Texture {
        return t
    }

    fun setTexture(texture: Texture) {
        t = texture
    }
}
